File: src/main/java/legacy/kms/chapter02/discount/AmountDiscountPolicy.java
package legacy.kms.chapter02.discount;

import legacy.kms.chapter02.Money;
import legacy.kms.chapter02.Screening;

public class AmountDiscountPolicy extends DefaultDiscountPolicy {

    private Money discountAmount;

    public AmountDiscountPolicy(Money discountAmount, DiscountCondition ... conditions) {
        super(conditions);
        this.discountAmount = discountAmount;
    }

    @Override
    protected Money getDiscountAmount(Screening screening) {
        return discountAmount;
    }
}

File: src/main/kotlin/kms/chapter02/discount/AmountDiscountPolicy.kt
package kms.chapter02.discount

import kms.chapter02.Money
import kms.chapter02.Screening

internal class AmountDiscountPolicy(
    private val discountAmount: Money,
    conditions: List<DiscountCondition>,
) : DefaultDiscountPolicy(conditions) {

    constructor(discountAmount: Money, vararg conditions: DiscountCondition) : this(discountAmount, conditions.toList())

    override fun getDiscountAmount(screening: Screening): Money {
        return discountAmount
    }
}
